src/instructlab/training/utils.py

```diff
@@ -624,71 +624,114 @@ def _copy_no_lora_dict(state_dict):
     return cleaned_state_dict
 
 
-def save_hf_format_ds(
+def save_dict_accelerate(
+    accelerator,
+    state_to_save,
+    save_directory,
+    max_shard_size="10GB",
+    safe_serialization=True,
+):
+    old_get_state = accelerator.get_state_dict
+    accelerator.get_state_dict = _copy_no_lora_dict
+
+    def skip_precheck_loops():
+        return []
+
+    # The save model does a loop over modules and params in order to determine how to get state dict. Since we already have the state dict directly, we want to bypass those checks.
+    state_to_save.modules = skip_precheck_loops
+    state_to_save.parameters = skip_precheck_loops
+
+    accelerator.save_model(
+        state_to_save,
+        save_directory=save_directory,
+        max_shard_size=max_shard_size,
+        safe_serialization=safe_serialization,
+    )
+
+    accelerator.get_state_dict = old_get_state
+
+
+def save_hf_format_accelerate(
     args,
     model,
     tokenizer,
+    accelerator,
```

Comment:
I recommend type-hinting this:


```suggestion
    accelerator: Accelerator,
```

Reply:
Fixed locally